### PR TITLE
[otel-agent] DD extension implementation 

### PR DESCRIPTION
**Pull request by truthbk**
### What does this PR do?

This PR introduces the DD extensions which will allow interfacing the otel-agent with the core agent and rest of the DD ecosystem. 

### Motivation

Add support for several features moving forward for the otel-agent and seamlessly integrate into the agent ecosystem and workflow.

### CODE REVIEW

File: comp/otelcol/extension/def/go.mod [hunk 1]
```diff
@@ -0,0 +1,30 @@
+module github.com/DataDog/datadog-agent/comp/otelcol/extension/def
+
+go 1.21.7
```
Comment (liustanley) [inline comp/otelcol/extension/def/go.mod:3]:
```suggestion
go 1.21.0
```

File: comp/metadata/inventoryotel/inventoryotelimpl/inventoryotel.go [hunk 2]
```diff
@@ -99,14 +100,20 @@
 
 func newInventoryOtelProvider(deps dependencies) (provides, error) {
 	hname, _ := hostname.Get(context.Background())
+	// HTTP client need not verify otel-agent cert since it's self-signed
+	// at start-up. TLS used for encryption not authentication.
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
```
[the remark this answers is not in the document]
Comment (truthbk) [reply comp/metadata/inventoryotel/inventoryotelimpl/inventoryotel.go:106]: This should be OK, we will only be listening on localhost on the extension side (still to be enforced), and TLS is used for encryption only (authorization via bearer token in follow-up work).

**Review by vickenty — APPROVED** [review]

File: comp/otelcol/extension/def/go.mod [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+	github.com/mitchellh/copystructure v1.2.0 // indirect
+	github.com/mitchellh/reflectwalk v1.0.2 // indirect
+	go.opentelemetry.io/collector/component v0.102.1 // indirect
+	go.opentelemetry.io/collector/config/configtelemetry v0.102.1 // indirect
+	go.opentelemetry.io/collector/confmap v0.102.1 // indirect
+	go.opentelemetry.io/collector/pdata v1.9.0 // indirect
+	go.opentelemetry.io/otel v1.27.0 // indirect
+	go.opentelemetry.io/otel/metric v1.27.0 // indirect
+	go.opentelemetry.io/otel/trace v1.27.0 // indirect
+	go.uber.org/multierr v1.11.0 // indirect
```
Comment (dineshg13) [inline comp/otelcol/extension/def/go.mod:22]: Use latest collector version modules. We have updated to 103 in PR https://github.com/DataDog/datadog-agent/pull/26943

File: comp/otelcol/extension/fx/fx.go [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+// Package extensionfx provides fx access for the provider component
+package extensionfx
+
+import (
+	extensionimpl "github.com/DataDog/datadog-agent/comp/otelcol/extension/impl"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
```
Comment (dineshg13) [inline comp/otelcol/extension/fx/fx.go:11]: Should we name it as flare extension? There might be other extensions we might add in future

Comment (truthbk) [reply comp/otelcol/extension/fx/fx.go:11]: I don't have a problem with that, but do you mind if we do this in a follow-up PR?

Comment (dineshg13) [reply comp/otelcol/extension/fx/fx.go:11]: sure that works.

**Review by dineshg13 — APPROVED** [review]
LGTM. @mackjmr Can u please review converter section.

**Review by mackjmr — APPROVED** [review]
> Can u please review converter section.

LGTM, although the logic to provide the configs will move out of the converter once https://github.com/DataDog/datadog-agent/pull/26967 is done.

**Review by dustmop — APPROVED** [review]
LGTM for agent-shared-components

**Review by ofek — APPROVED** [review]

**Review by robertjli — APPROVED** [review]

**Review by jszwedko — APPROVED** [review]
👍 for files owned by @DataDog/agent-processing-and-routing

File: tasks/modules.py [hunk 5]
```diff
@@ -151,6 +151,8 @@ def dependency_path(self, agent_version):
     ),
     "comp/logs/agent/config": GoModule("comp/logs/agent/config", independent=True, used_by_otel=True),
     "comp/netflow/payload": GoModule("comp/netflow/payload", independent=True),
+    "comp/otelcol/extension/def": GoModule("comp/otelcol/extension/def", independent=True, used_by_otel=True),
+    "comp/otelcol/extension/impl": GoModule("comp/otelcol/extension/impl", independent=True, used_by_otel=True),
```
Comment (CelianR) [inline tasks/modules.py:155]: **🥜 nitpick:** ‏Could you add these modules after `comp/otelcol/collector-contrib/def` to keep the order ?

**Review by CelianR — APPROVED** [review]

**Review by CelianR — APPROVED** [review]

**Review by KevinFairise2 — APPROVED** [review]

**Review by ichinaski — APPROVED** [review]
LGTM from trace-agent 👍

**Review by pducolin — COMMENTED** [review]
ok for files owned by @DataDog/agent-devx-loops

**Pull request merged**